state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -1965,6 +1969,284 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
+	) -> Option<BoostStatus<TargetChainAmount<T, I>>> {
+		if amount < MinimumDeposit::<T, I>::get(asset) {
+			// We do not process/record pre-witnessed deposits for amounts smaller
+			// than MinimumDeposit to match how this is done on finalisation
+			return None;
+		}
+
+		if let Some(tx_id) = deposit_details.deposit_id() {
+			if TaintedTransactions::<T, I>::mutate(broker, &tx_id, |opt| {
+				match opt.as_mut() {
+					// Transaction has been reported, mark it as pre-witnessed.
+					Some(status @ TaintedTransactionStatus::Unseen) => {
+						*status = TaintedTransactionStatus::Prewitnessed;
+						true
+					},
+					// Pre-witnessing twice is unlikely but possible. Either way we don't want
+					// to change the status and we don't want to allow boosting.
+					Some(TaintedTransactionStatus::Prewitnessed) => true,
+					// Transaction has not been reported, mark it as boosted to prevent further
+					// reports.
+					None => false,
+				}
+			}) {
+				return None;
+			}
+		}
+
+		let prewitnessed_deposit_id = PrewitnessedDepositIdCounter::<T, I>::mutate(|id| -> u64 {
+			*id = id.saturating_add(1);
+			*id
+		});
+
+		// Only boost on non-zero fee and if the channel isn't already boosted:
+		if T::SafeMode::get().boost_deposits_enabled &&
+			boost_fee > 0 &&
+			!matches!(boost_status, BoostStatus::Boosted { .. })
+		{
+			match Self::try_boosting(asset, amount, boost_fee, prewitnessed_deposit_id) {
+				Ok(BoostOutput { used_pools, total_fee: boost_fee_amount }) => {
+					let amount_after_boost_fee = amount.saturating_sub(boost_fee_amount);
+
+					// Note that ingress fee is deducted at the time of boosting rather than the
+					// time the deposit is finalised (which allows us to perform the channel
+					// action immediately):
+					let AmountAndFeesWithheld { amount_after_fees, fees_withheld: ingress_fee } =
+						Self::conditionally_withhold_ingress_fee(
+							asset,
+							amount_after_boost_fee,
+							&origin,
+						);
+
+					let used_pool_tiers = used_pools.keys().cloned().collect();
+
+					if let Ok(action) = Self::perform_channel_action(
+						action,
+						asset,
+						source_address,
+						amount_after_fees,
+						origin.clone(),
+					) {
+						Self::deposit_event(Event::DepositBoosted {
+							deposit_address,
+							asset,
+							amounts: used_pools,
+							block_height,
+							prewitnessed_deposit_id,
+							channel_id,
+							deposit_details: deposit_details.clone(),
+							ingress_fee,
+							boost_fee: boost_fee_amount,
+							action,
+							origin_type: origin.into(),
+						});
+					} else {
+						// TODO: emit error?
+					}
+
+					return Some(BoostStatus::Boosted {
+						prewitnessed_deposit_id,
+						pools: used_pool_tiers,
+						amount,
+					});
+				},
+				Err(err) => {
+					Self::deposit_event(Event::InsufficientBoostLiquidity {
+						prewitnessed_deposit_id,
+						asset,
+						amount_attempted: amount,
+						channel_id,
+					});
+					log::debug!(
```

Comment:
I don't think we need a log if we have an event

Reply:
removed